crates/runtime/examples/crdt.rs

```diff
@@ -0,0 +1,237 @@
+use calimero_storage::address::{Id, Path as EntityPath};
+use calimero_storage::entities::{ChildInfo, Element};
+use calimero_storage::interface::Action;
+use eyre::Result as EyreResult;
+use owo_colors::OwoColorize;
+use serde_json::json;
+
+fn main() -> EyreResult<()> {
+    let args: Vec<String> = env::args().collect();
+    if args.len() != 2 {
+        println!("Usage: {args:?} <path-to-wasm>");
+        return Ok(());
+    }
+
+    let path = &args[1];
+    let path = Path::new(path);
+    if !path.exists() {
+        eyre::bail!("KV wasm file not found");
+    }
+
+    let file = File::open(path)?.bytes().collect::<Result<Vec<u8>, _>>()?;
+
+    let mut storage = InMemoryStorage::default();
+
+    let limits = VMLimits::new(
+        /*max_stack_size:*/ 200 << 10, // 200 KiB
+        /*max_memory_pages:*/ 1 << 10, // 1 KiB
+        /*max_registers:*/ 100,
+        /*max_register_size:*/ (100 << 20).validate()?, // 100 MiB
+        /*max_registers_capacity:*/ 1 << 30, // 1 GiB
+        /*max_logs:*/ 100,
+        /*max_log_size:*/ 16 << 10, // 16 KiB
+        /*max_events:*/ 100,
+        /*max_event_kind_size:*/ 100,
+        /*max_event_data_size:*/ 16 << 10, // 16 KiB
+        /*max_storage_key_size:*/ (1 << 20).try_into()?, // 1 MiB
+        /*max_storage_value_size:*/ (10 << 20).try_into()?, // 10 MiB
+    );
+
+    let cx = VMContext::new(Vec::new(), [0; 32]);
+
+    let _outcome = run(&file, "init", cx, &mut storage, &limits)?;
+    // dbg!(&outcome);
+
+    // Set up Library
+    println!("{}", "--".repeat(20).dimmed());
+    println!("{:>35}", "Setting up Library".bold());
+    println!("{}", "--".repeat(20).dimmed());
+    
+    #[derive(BorshSerialize)]
+    pub struct Library {
+        books: Books,
+        storage: Element,
+    }
+    #[derive(BorshSerialize)]
+    pub struct Books;
+    
+    // let library_id: [u8; 16] =
+    //     hex::decode("deadbeef-1122-3344-5566-778899aabb01".replace("-", ""))?
+    //         .as_slice()
+    //         .try_into()?;
+    // let library_data = json!({
+    //     "books": [],
+    //     "storage": {
+    //         "id": hex::encode(library_id),
+    //         "is_dirty": false,
+    //         "merkle_hash": hex::encode([0; 32]),
+    //         "metadata": {
+    //             "created_at": 0,
+    //             "updated_at": 0
+    //         },
+    //         "path": "::library"
+    //     }
+    // });
+    let library_data = Library {
+        books: Books {},
+        storage: Element::new(&EntityPath::new("::library")?),
+    };
+    let library_id = library_data.storage.id();
+    
+    let action = Action::Add {
+        id: Id::from(library_id),
+        type_id: 11,
+        // data: serde_json::to_vec(&library_data)?,
+        data: borsh::to_vec(&library_data)?,
+        ancestors: Vec::new(),
+    };
+    let serialized_action = serde_json::to_string(&action)?;
+    let input = std::fmt::format(format_args!("{{\"action\": {}}}", serialized_action));
```

Comment:
```suggestion
    let input = format!("{{\"action\": {}}}", serialized_action);
```

or better yet

```suggestion
    let input = json!({ "action": serialized_action });
```

Reply:
No issues there - but I will hold off from making any changes here given that I'm not currently working on the examples. I don't want to overlap. The examples are pretty raw and are rough test code only. This PR is still in draft. Ideally review at this stage is for overall approach and direction of changes, and not things at a line level.